operator/controllers/kyma_controller.go

```diff
@@ -418,16 +410,35 @@ func (r *KymaReconciler) CreateModule(ctx context.Context, name string, kyma *v1
 func (r *KymaReconciler) UpdateModule(ctx context.Context, name string, kyma *v1alpha1.Kyma,
 	module *parsed.Module,
 ) error {
+	if module.Template.Spec.Target == v1alpha1.TargetControlPlane {
+		return nil
+	}
+
+	if err := r.setupModule(module, kyma, name); err != nil {
+		return err
+	}
+
+	if err := r.Update(ctx, module.Unstructured, &client.UpdateOptions{}); err != nil {
+		return fmt.Errorf("error updating custom resource of type %s %w", name, err)
+	}
+
+	return nil
+}
+
+func (r *KymaReconciler) setupModule(module *parsed.Module, kyma *v1alpha1.Kyma, name string) error {
 	// merge template and component settings
 	if err := module.CopySettingsToUnstructured(); err != nil {
 		return fmt.Errorf("error occurred while updating module from settings: %w", err)
 	}
 	// set labels
 	module.ApplyLabels(kyma, name)
-	// update the spec
-	module.Unstructured.Object["spec"] = module.Template.Spec.Data.Object["spec"]
-	if err := r.Update(ctx, module.Unstructured, &client.UpdateOptions{}); err != nil {
-		return fmt.Errorf("error updating custom resource of type %s %w", name, err)
+
+	if module.GetOwnerReferences() == nil {
```

Comment:
this will break if there is an owner reference that comes from a different controller, maybe we should introduce an error case here

Reply:
add a new function `ContainsExpectedOwnerReference` to check if contains kyma as owner. btw, we allow other ownerreference exists, right?